### ISSUE TITLE
Include additional properties in ProcessError

### ISSUE THREAD

**Issue by iforapsy** [ronf/asyncssh#106]
First of all, thank you for all your work on asyncssh. This library is easy to use, and the documentation is top-notch.

I would like `asyncssh.process.ProcessError` to include `stdout` and `stderr`. Most of the time, the exit code of the remote process is not enough to determine how to handle a ProcessError. I need to inspect the standard output and standard error. For example, this is how I would ideally handle errors.
```
try:
    await connection.run('rm my_file', check=True)
except asyncssh.process.ProcessError as ex:
    if 'Permission denied' in ex.stderr:
        logger.warn('Need more permissions')
    elif 'No such file or directory' in ex.stderr:
        # Ignore error since file is already gone
        pass
    else:
        raise
```
Right now, I have to use `check=False` so that I can inspect the standard error on the returned `SSHCompletedProcess`. It works, but it's not Pythonic since it doesn't use exceptions.

If you have time, it would be handy for `asyncssh.process.ProcessError` to have all of [`subprocess.CalledProcessError`](https://docs.python.org/3/library/subprocess.html#subprocess.CompletedProcess)'s properties like `returncode` and `cmd`. There's a lot of code out there that uses `subprocess` and handles `subprocess.CalledProcessError`. If `asyncssh.process.ProcessError` matches that class's API, it would be easier for people to port their local subprocess code to run remotely.

**Comment by ronf** [comment]
I'm glad to hear that asyncssh is working well for you, and really appreciate your suggestion! I think it's a good idea, and I believe it should be a fairly straightforward change. I'll take a closer look at it this weekend.

Regarding "returncode", I looked into providing that when I originally built this, but ran into a problem getting numeric values for signals, as they vary from one OS to another. When getting back a signal from another host via SSH, I'll have the signal name, but I may not be able to map that to a signal number on the client machine to put in "returncode". So, I settled for returning an "exit_status" of -1 in all cases where a signal was returned, and setting a separate "exit_signal" value with the string representation of which signal was returned.

I should be able to provide "stdout", and "stderr" without a problem. Providing "cmd" looks like it will be a bit more effort, as I currently don't save that value in the client channel or client session. It's just passed to the remote system and then forgotten right now. I'll see what I can do on that, though.

**Comment by ronf** [comment]
Ok - I've just checked in 12fe614 on the AsyncSSH "develop" branch which adds this functionality.

Specifically, in addition to adding "stdout" and "stderr" to ProcessError, I have added "env", "command", and "subsystem" properties to SSHClientProcess, SSHServerProcess, SSHCompletedProcess, and ProcessError.

In terms of naming, I kept things consistent with the argument names used in AsyncSSH to set these values, rather than trying to exactly mimic subprocess.CalledProcessError. Part of the reason for that is that the subprocess module isn't very consistent. For instance, it uses "cmd" in CalledProcessError, but it uses "args" in CompletedProcess (and that can be either a string or a list). With the properties I've added, it should be pretty simple to port existing subprocess code. The name changes can be done at the same time as other necessary changes like adding "await".

**Comment by iforapsy** [comment]
@ronf Thank you. I appreciate your prompt response and code change.

**Comment by ronf** [comment]
These changes are now available in the 1.11.0 release I just posted.

**Comment by ronf** [comment]
I know this has been closed a while, but as part of the work on #188 I have now implemented a 'returncode' property on SSHClientProcess, SSHCompletedProcess, and ProcessError. There's also a corresponding get_returncode() call on SSHClientChannel. I solved the signal numbering problem by using the signal names and numbers defined in the local system's "signal" module to map the text names of signals coming over SSH back to the local numeric equivalent. In cases where an exit signal is reported by a remote system that has no local equivalent, the return code is set to -99.

As part of this change, I also now allow either text or numeric signal values to be passed in calls to send_signal() as well. Numeric signal values much have a corresponding signal name defined in the "signal" module, and the value is converted to text before being sent over SSH. If there's no match for the signal number, ValueError is raised.
